retrofit-adapters/rxjava/src/main/java/retrofit/RxJavaCallAdapterFactory.java

```diff
@@ -173,13 +182,8 @@ private CallOnSubscribe(Call<T> originalCall) {
       return responseType;
     }
 
-    @Override public <R> Observable<Result<R>> adapt(Call<R> call) {
-      return Observable.create(new CallOnSubscribe<>(call)) //
-          .map(new Func1<Response<R>, Result<R>>() {
-            @Override public Result<R> call(Response<R> response) {
-              return Result.response(response);
-            }
-          })
+    @Override public <R> Observable<Result<R>> adapt(Call<R> call, Retrofit retrofit) {
+      return Observable.create(new CallOnSubscribe<>(call, retrofit)) //
           .onErrorReturn(new Func1<Throwable, Result<R>>() {
```

Comment:
I thought that initially, but that would require us calling `onNext(Result.error())` in `CallOnSubscribe`, which means that for the other adapters we'd have to purposefully throw that error later.